common/workunit/workunit.cpp

```diff
@@ -11568,28 +11568,45 @@ extern WORKUNIT_API StringBuffer &exportWorkUnitToXML(const IConstWorkUnit *wu,
         return str.append("Unrecognized workunit format");
 }
 
-extern WORKUNIT_API void exportWorkUnitToXMLFile(const IConstWorkUnit *wu, const char * filename, unsigned extraXmlFlags, bool unpack, bool includeProgress, bool hidePasswords, bool splitStats)
+extern WORKUNIT_API void exportWorkUnitToXMLFile(const IConstWorkUnit *wu, const char * filename, unsigned extraXmlFlags, bool unpack, bool includeProgress, bool hidePasswords, bool regressionTest)
 {
     const IExtendedWUInterface *ewu = queryExtendedWU(wu);
     if (ewu)
     {
         Linked<IPropertyTree> p;
         if (unpack||includeProgress)
             p.setown(ewu->getUnpackedTree(includeProgress));
+        else if (regressionTest)
+            p.setown(createPTreeFromIPT(ewu->queryPTree()));
         else
             p.set(ewu->queryPTree());
         if (hidePasswords)
             return exportWorkUnitToXMLFileWithHiddenPasswords(p, filename, extraXmlFlags);
-        if (splitStats)
+
+        if (regressionTest)
         {
-            StringBuffer statsFilename;
-            statsFilename.append(filename).append(".stats");
+            //This removes any items from the xml that will vary from run to run, so they can be binary compared from run to run
+            //The following attributes change with the build.  Simpler to remove rather than needing to updated each build
+            p->removeProp("@buildVersion");
+            p->removeProp("@eclVersion");
+            p->removeProp("@hash");
+
```

Comment:
I don't see it removing any 'ts' attributes or statistic  types such as WhenStarted and TimeElapsed.  For a binary compare should these be removed to.  I'm referring the stats output in the following format:
<Statistics>
 <Statistic c="eclcc" count="1" creator="eclcc" kind="WhenStarted" s="compile" scope="compile" **ts="1541063347021154"** **unit="ts"** **value="1541063347021131"**/>
 <Statistic c="eclcc" count="1" creator="eclcc" kind="WhenStarted" s="compile" scope="compile:parse" **ts="1541063347021212" **unit="ts"** **value="1541063347021202"**/>
 <Statistic c="eclcc" count="1" creator="eclcc" kind="TimeElapsed" s="compile" scope="compile:parse" **ts="1541063353737173" **unit="ns"** **value="6719330841"**/>
 <Statistic c="eclcc" count="1" creator="eclcc" kind="NumAttribsSimplified" s="compile" scope="compile:cache" **ts="1541063353737203"** unit="cnt" value="0"/>
 <Statistic c="eclcc" count="1" creator="eclcc" kind="NumAttribsProcessed" s="compile" scope="compile:cache" **ts="1541063353737233"** unit="cnt" value="250"/>
 <Statistic c="eclcc" count="1" creator="eclcc" kind="NumAttribsFromCache" s="compile" scope="compile:cache" **ts="1541063353737264"** unit="cnt" value="0"/>
 <Statistic c="eclcc" count="1" creator="eclcc" kind="NumAttribsSimplifiedTooComplex" s="compile" scope="compile:cache" **ts="1541063353737301"** unit="cnt" value="0"/>
</Statistics>

Reply:
All of the statistics are removed - see the next lines, and extracted to a separate file.